test/properties_test.exs

```diff
@@ -706,4 +706,61 @@ defmodule Surface.PropertiesSyncTest do
              code.exs:7:\
            """
   end
+
+  test "warn if given default value doesn't exist in values list" do
+    id = :erlang.unique_integer([:positive]) |> to_string()
+    module = "TestComponentWithDefaultValueThatDoesntExistInValues_#{id}"
+
+    code = """
+    defmodule #{module} do
+      use Surface.Component
+
+      prop type, :string, values!: ["small", "medium", "large"], default: "x-large"
+
+      data data_type, :string, values!: ["small", "medium", "large"], default: "x-large"
+
+      prop invalid_type, :integer, default: [], values!: [0, 1, 2]
+
+      prop valid_acc, :integer, default: [1], values!: [0, 1, 2], accumulate: true
+
+      prop invalid_acc1, :integer, default: [3], values!: [0, 1, 2], accumulate: true
+
+      prop invalid_acc2, :string, values!: [1, 2, 3], default: 3, accumulate: true
+
+      def render(assigns) do
+        ~H""
+      end
+    end
+    """
+
+    output =
+      capture_io(:standard_error, fn ->
+        {{:module, _, _, _}, _} =
+          Code.eval_string(code, [], %{__ENV__ | file: "code.exs", line: 1})
+      end)
+
+    assert output =~ ~S"""
+           prop `type` default value "x-large" does not exist in :values!
+           """
+
+    assert output =~ ~S"""
+           data `data_type` default value "x-large" does not exist in :values!
+           """
+
+    assert output =~ ~S"""
+           prop `invalid_type` default value [] does not exist in :values!
+           """
+
+    refute output =~ ~S"""
+           prop `valid_acc`
+           """
+
+    assert output =~ ~S"""
+           prop `invalid_acc1` default value [3] does not exist in :values!
+           """
+
+    assert output =~ ~S"""
+           prop `invalid_acc2` default value 3 must be a list when accumulate: true
```

Comment:
should i wrap the value in "``" like i do for the prop/data name?

Reply:
Yeah, I think we should start using  \`  \`  whenever we have code-related stuff, just like we do in markdown. If in the future, we want to turn those boring monochromatic messages into beautiful colourful ones (like in Rust), it won't have to update any of the messages.

Maybe this is also valid for https://github.com/surface-ui/surface/pull/374/files#diff-fb7dcaf7105c49bd4b887f35e2c32958116d06d3ef6639370ed6cb3741f287cbR491 for `accumulate: true`.